packages/common/src/utils/convertCustomDimensionsToYaml.ts

```diff
@@ -1,12 +1,63 @@
-import type { DbtColumnLightdashDimension } from '../types/dbt';
-import { type CustomSqlDimension, friendlyName } from '../types/field';
+import type { DbtColumnLightdashAdditionalDimension } from '../types/dbt';
+import { ParameterError } from '../types/errors';
+import {
+    BinType,
+    type CustomBinDimension,
+    type CustomSqlDimension,
+    DimensionType,
+    friendlyName,
+} from '../types/field';
+import { type WarehouseClient } from '../types/warehouse';
+import {
+    getCustomRangeSelectSql,
+    getFixedWidthBinSelectSql,
+} from './customDimensions';
 
 export const convertCustomDimensionToDbt = (
     field: CustomSqlDimension,
-): DbtColumnLightdashDimension => ({
+): DbtColumnLightdashAdditionalDimension => ({
     label: friendlyName(field.name),
-    name: field.id,
     description: '', // TODO :: leave it here so that customer can easily fill in?
     type: field.dimensionType,
     sql: field.sql,
 });
+
+export const convertCustomBinDimensionToYaml = ({
+    customDimension,
+    baseDimensionSql,
+    warehouseClient,
+}: {
+    customDimension: CustomBinDimension;
+    baseDimensionSql: string;
+    warehouseClient: WarehouseClient;
+}): DbtColumnLightdashAdditionalDimension => {
+    switch (customDimension.binType) {
+        case BinType.CUSTOM_RANGE:
+            return {
+                label: friendlyName(customDimension.name),
+                description: '',
```

Comment:
I've removed empty description in my PR, schema requires it to have a length of at least 8 characters.

![CleanShot 2025-03-05 at 10 50 22@2x](https://github.com/user-attachments/assets/8b0c04e1-746b-4165-be77-adc32181603b)

If you open a PR, merge it, refresh dbt and try to write-back another one, YML validation fails during the read. Maybe we should validate the output file as well?